Latest release causes dump-error on some packages

I am seeing the following error when running my testbed with the -n flag on py.test with xdist 1.19.0, downgrading to xdist 1.18.2 fixes the issue 

```
_____________________________ ERROR collecting tests/unit/featureflag/test_pyrollout.py ______________________________
<remote exec>:102: in pytest_logwarning
    ???
<remote exec>:26: in sendevent
    ???
../../.virtualenvs/lead-pages/lib/python2.7/site-packages/execnet/gateway_base.py:717: in send
    self.gateway._send(Message.CHANNEL_DATA, self.id, dumps_internal(item))
../../.virtualenvs/lead-pages/lib/python2.7/site-packages/execnet/gateway_base.py:1354: in dumps_internal
    return _Serializer().save(obj)
../../.virtualenvs/lead-pages/lib/python2.7/site-packages/execnet/gateway_base.py:1372: in save
    self._save(obj)
../../.virtualenvs/lead-pages/lib/python2.7/site-packages/execnet/gateway_base.py:1390: in _save
    dispatch(self, obj)
../../.virtualenvs/lead-pages/lib/python2.7/site-packages/execnet/gateway_base.py:1475: in save_tuple
    self._save(item)
../../.virtualenvs/lead-pages/lib/python2.7/site-packages/execnet/gateway_base.py:1390: in _save
    dispatch(self, obj)
../../.virtualenvs/lead-pages/lib/python2.7/site-packages/execnet/gateway_base.py:1471: in save_dict
    self._write_setitem(key, value)
../../.virtualenvs/lead-pages/lib/python2.7/site-packages/execnet/gateway_base.py:1465: in _write_setitem
    self._save(value)
../../.virtualenvs/lead-pages/lib/python2.7/site-packages/execnet/gateway_base.py:1388: in _save
    raise DumpError("can't serialize %s" % (tp,))
E   DumpError: can't serialize <class 'py._path.local.LocalPath'>
```

--- a/changelog/214.bugfix
+++ b/changelog/214.bugfix
@@ -1,0 +1,1 @@
+Fix crash when transferring internal pytest warnings from workers to the master node.

--- a/testing/acceptance_test.py
+++ b/testing/acceptance_test.py
@@ -340,18 +340,28 @@ class TestTerminalReporting:
         ])
 
     @pytest.mark.parametrize('n', ['-n0', '-n1'])
-    def test_logwarning(self, testdir, n):
+    @pytest.mark.parametrize('warn_type', ['pytest', 'builtin'])
+    def test_logwarning(self, testdir, n, warn_type):
         from pkg_resources import parse_version
         if parse_version(pytest.__version__) < parse_version('3.1'):
             pytest.skip('pytest warnings requires >= 3.1')
+
+        if warn_type == 'builtin':
+            warn_code = """warnings.warn(UserWarning('this is a warning'))"""
+        elif warn_type == 'pytest':
+            warn_code = """request.config.warn('', 'this is a warning',
+                           fslocation=py.path.local())"""
+        else:
+            assert False
         testdir.makepyfile("""
-            import warnings
-            def test_func():
-                warnings.warn('this is a warning')
-        """)
+            import warnings, py
+            def test_func(request):
+                {warn_code}
+        """.format(warn_code=warn_code))
         result = testdir.runpytest(n)
         result.stdout.fnmatch_lines([
             "*this is a warning*",
+            "*1 passed, 1 warnings*",
         ])
 
 

--- a/xdist/remote.py
+++ b/xdist/remote.py
@@ -99,7 +99,7 @@ class SlaveInteractor:
 
     def pytest_logwarning(self, message, code, nodeid, fslocation):
         self.sendevent("logwarning", message=message, code=code, nodeid=nodeid,
-                       fslocation=fslocation)
+                       fslocation=str(fslocation))
 
 
 def serialize_report(rep):

